### ISSUE TITLE
[no-unnecessary-condition] False negative when nullable type is on the left hand side

### ISSUE THREAD

**Issue by jzarzeckis** [typescript-eslint/typescript-eslint#1308]
**Repro**

```JSON
{
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": [1, { "ignoreRhs": true }],
  }
}
```

```TS
function fun(foo: string[] | undefined): string[] {
  const bar: string[] = foo || ['hello']

  return bar
}
```

**Expected Result**
No warnings

**Actual Result**
```TS
function fun(foo: string[] | undefined): string[] {
  const bar: string[] = foo || ['hello']
  //                    ^^^ Unnecessary conditional, value is always truthy.eslint(@typescript-eslint/no-unnecessary-condition)
  return bar
}
```

![image](https://user-images.githubusercontent.com/919350/70248557-bac74e80-1783-11ea-8d60-c205909a2731.png)

**Additional Info**
`strict` mode has been enabled in `tsconfig`. I mean I'd expect such behaviour if strict null checks were disabled, but not when they are enabled.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `5.9.0` |
| `node`                             | `13.1.0` |
| `npm`                              | `6.12.1` |

**Comment by glen-84** [comment]
Did you reload VS Code after updating `@typescript-eslint/*`?

You may also want to consider updating `eslint`.

**Comment by jzarzeckis** [comment]
Sorry for the hassle.
Now that I've upgraded **everything** - the issue is no longer reproduced. It can be closed

**Comment by jzarzeckis** [comment]
I guess the problem was that I kept doing "Disable/Enable Eslint" while a full reload was required.
